### DISCUSSION TITLE
Venn/Euler Diagrams

### DISCUSSION THREAD

**Discussion by lloydjatkinson** [observablehq/plot#2199] [Ideas]
There's a real lack of reusable venn/euler diagram libraries. It would be awesome if this library added it.

**Comment by Fil** [comment]
Have you tried https://github.com/benfred/venn.js ?

**Reply by lloydjatkinson** [reply]
Yes and it’s why I made this discussion 😅.

The author appears no longer active on GitHub and all of their libraries including Venn.js are unmaintained. The fork of Venn.js also seems to have become unmaintained. There’s a discussion here: https://github.com/benfred/fmin/pull/16

I think it would be a great feature for Observable to have!

**Reply by Fil** [reply]
Thanks. I'm also using **fmin** in several projects.

The library might not have been updated for a long time, but it's still working very well as is:
https://observablehq.com/@fil/venn-js

Only caveat is that it's loading its own copy of d3 in a rather older version.

It would be interesting to couple it with Plot somehow — probably by returning the circles as geometries and using Plot.geo to draw them.
